### DISCUSSION TITLE
Guardian UI Tracker

### DISCUSSION THREAD

**Discussion by garyray-k** [fedimint/fedimint#2361] [General]
Building from #2337 but the aim for this Issue is to track overall completeness of the Guardian UI.

Tasks under UI Progress are for creating the actual components, but there may be a stub for the actual functionality when interacting with the mint over the Config API. These tasks aim to cover all "screens" shown in the [Figma](https://www.figma.com/file/ioeAVbWBc3ow2RaZETatM7/Fedimint-Setup-Flow?node-id=0-1&t=lWkqGrDVvibwsv1I-0), referred to by their Figma name for reference sake. The Config API and Other section is tasks involving the API layer of the UI and general patterns for across the app. [Here is another example with more of the API layer implemented](https://github.com/justinmoon/fedimint/commit/7fa9b1d475a1e2904500885f6942022df9811117)

**UI Progress** 

- [x] Select Guardian Role
- [x] Leader: Setup Federation
- [x] Leader: Invite Guardians
  - [x] Placeholder
  - [x] Experience
- [x] Follower: Setup Federation
- [x] Follower: Connect with Leader
  - [x] Placeholder
  - [x] Experience
- [x] Both: Run Keygen ?
- [x] Both: Verify Guardians (https://github.com/okjodom/fedimint/pull/126)
  - [x] Placeholder
  - [x] Experience
- [x] Both: Guardian Dashboard
  - [x] Placeholder
  - [x] Experience
- [x] General style sheet (based on https://www.bitcoinuikit.com/ and  [Figma](https://www.figma.com/file/ioeAVbWBc3ow2RaZETatM7/Fedimint-Setup-Flow))
- [x] Reusable Forms/components (might just be accomplished already with Chakra) (https://github.com/fedimint/fedimint/issues/2429)

**Config API and Other**

- [x] Websocket connection 
- [x] Loading State/Reactivity (create a pattern to standardize across all components)
- [x] App (not Config) State management
- [ ] Error handling
- [ ] Integration tests
- [x] App Navigation (Routes, etc)

my two sats: I think vanilla React useContext/Reducer should work well for this size of app (#2345). We'll have to couple that with Routes so a non-technical user gets the standard Back button experience. It seems the web socket should be a singleton, but I'm not familiar with that pattern in React/in general. Chakra seems to work fine. I'm not sure many more packages should be needed for MVP. Screenshots and test instructions will be super helpful for the visual and flow of the app. API logic and reactivity should ideally have unit tests. Integration tests can come later.

**Comment by garyray-k** [comment]
https://github.com/fedimint/fedimint/issues/2307

Posting here for reference.

**Comment by garyray-k** [comment]
Somewhat related:
https://github.com/fedimint/fedimint/issues/2294

**Comment by garyray-k** [comment]
@wbobeirne 
Seems like most things in the original list were covered. I'd think the unit/integration tests would still be needed as well as consolidated error handling.
Thoughts?

**Comment by okjodom** [comment]
Now tracking all UI work in https://github.com/fedimint/ui . Closing this initial tracker
